classes/package.php

```diff
@@ -230,41 +234,30 @@ public function store_in_db(int $contextid = 0) {
                 "tag" => $tag,
             ];
         }
+
         $transaction = $DB->start_delegated_transaction();
-        try {
-            $DB->insert_records('qtype_questionpy_tags', $tagsdata);
-            $DB->insert_records('qtype_questionpy_language', $languagedata);
-        } catch (\dml_exception $e) {
-            $DB->rollback_delegated_transaction($transaction, $e);
-        }
-        $DB->commit_delegated_transaction($transaction);
+        $DB->insert_records('qtype_questionpy_tags', $tagsdata);
+        $DB->insert_records('qtype_questionpy_language', $languagedata);
+        $transaction->allow_commit();
```

Comment:
Du verwendest jetzt zwei separate transaktionen, eine für `qtype_questionpy_package`, und eine für `_tags` und `_language`. Das Ziel müsste doch aber sein, dass entweder alle drei zusammen, oder gar kein Insert durchgeht, also sollten alle drei Inserts in einer Transaktion stattfinden.

Reply:
stimmt auch. Änderung in: [744a1c0](https://github.com/questionpy-org/moodle-qtype_questionpy/pull/23/commits/744a1c071563730a5902efd3a6011e932be89732)